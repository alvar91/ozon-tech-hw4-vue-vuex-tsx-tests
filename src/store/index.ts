import Vue from "vue";
import Vuex from 'vuex';
import { createVuexStore } from "vuex-simple";

// @ts-ignore
import { MainModule } from "./modules/main/main.ts";

Vue.use(Vuex);

const store = createVuexStore(new MainModule());

export default store;
