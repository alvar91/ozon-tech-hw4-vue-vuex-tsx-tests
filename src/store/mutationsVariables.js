export const FETCH_USERS_ACTION = "FETCH_USERS_ACTION";
export const FETCH_USERS = "FETCH_USERS";
export const SET_LOADING = "SET_LOADING";
export const SET_CURRENT_USER = "SET_CURRENT_USER";
export const SET_FILTER_KEYWORD = "SET_FILTER_KEYWORD";
export const SET_FORM_INPUT = "SET_FORM_INPUT";
export const SET_FORM_CHANGE = "SET_FORM_CHANGE";
export const SET_FORM_RADIO = "SET_FORM_RADIO";
