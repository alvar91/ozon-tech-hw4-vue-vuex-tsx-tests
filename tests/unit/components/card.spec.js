import Vuex from "vuex";
import { mount, createLocalVue } from "@vue/test-utils";

import Card from "@/components/card.vue";

import { FETCH_USERS_ACTION } from "@/store/mutationsVariables";

import store from "@/store/index";

const localVue = createLocalVue();
localVue.use(Vuex);

describe("Card.vue", () => {
  let wrapper;

  beforeEach(async () => {
    await store.dispatch(FETCH_USERS_ACTION);

    wrapper = mount(Card, {
      localVue,
      store,
    });
  });

  it("should correctly renders markup", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  afterEach(() => {
    wrapper.destroy();
  });
});
