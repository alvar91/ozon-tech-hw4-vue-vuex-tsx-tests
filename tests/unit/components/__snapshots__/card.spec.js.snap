// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card.vue should correctly renders markup 1`] = `
<main class="main">
  <section class="card">
    <article class="card__about">
      <h2 class="card__name">Агафонов Михаил Дмитриевич</h2><button class="card__user-status card__user-status--active">Загружена кандидатом</button>
    </article>
    <article class="card__data">
      <ul class="card__list">
        <li class="card__list-item">
          <ul class="card__sub-list">
            <li class="card__item"><span class="card__label">Заявка:</span><span class="card__label">Логин:</span></li>
            <li class="card__item"><span class="card__label">HR-48963</span><span class="card__label">mdagafonov</span></li>
          </ul>
        </li>
        <li class="card__list-item">
          <ul class="card__sub-list">
            <li class="card__item"><span class="card__label">Дата выхода:</span><span class="card__label">Дата оформления:</span></li>
            <li class="card__item"><time datetime="2020-08-03T08:54:38-03:00" class="card__date">03.08.2020</time><time datetime="2020-07-21T10:11:08-03:00" class="card__date">21.07.2020</time></li>
          </ul>
        </li>
        <li class="card__list-item">
          <ul class="card__sub-list">
            <li class="card__item"><span class="card__label">Статус ФП:</span><span class="card__label">Статус УЗ:</span></li>
            <li class="card__item"><span class="card__status card__status--active">Активна</span><span class="card__status card__status--success">Активна</span></li>
          </ul>
        </li>
      </ul>
    </article>
    <article class="card__detailed">
      <h3 class="card__personal">Личная информация</h3>
      <form action="#" class="card__form"><label class="card__field"><span class="card__title">Фамилия</span><input type="text" name="middleName" placeholder="Фамилия" class="card__input"></label><label class="card__field"><span class="card__title">Имя</span><input type="text" name="firstName" placeholder="Имя" class="card__input"></label><label class="card__field"><span class="card__title">Отчество</span><input type="text" name="lastName" placeholder="Отчество" class="card__input"></label><label class="card__field"><span class="card__title">Дата рождения</span><input type="date" name="age" min="1900-01-01" class="card__input card__input--date"></label><label class="card__field"><span class="card__title">Гражданство</span><select name="citizenship" class="card__select card__select--arrow">
            <option value="Грузия">Грузия</option>
            <option value="США">США</option>
            <option value="Российская Федерация">Российская Федерация</option>
            <option value="Украина">Украина</option>
            <option value="Беларусь">Беларусь</option>
            <option value="Казахстан">Казахстан</option>
          </select></label><span class="card__field"><span class="card__title">Пол</span><span class="card__label-container"><label tabindex="0" class="card__radio">Мужской<input type="radio" name="gender" class="card__input-radio" value="Мужской"><span class="card__checkmark"></span></label><label tabindex="0" class="card__radio">Женский<input type="radio" name="gender" class="card__input-radio" value="Женский"><span class="card__checkmark"></span></label></span></span><label class="card__field card__field--wide card__field--disabled"><span class="card__title">Город проживания</span><input type="text" name="city" placeholder="Город проживания" disabled="disabled" class="card__input"></label></form>
    </article>
  </section>
</main>
`;
